pkg/gadgets/trace/dns/types/dns.go

```diff
@@ -35,6 +35,9 @@ type Event struct {
 	PktType    string     `json:"pktType,omitempty" column:"type,minWidth:7,maxWidth:9"`
 	QType      string     `json:"qtype,omitempty" column:"qtype,minWidth:5,maxWidth:10"`
 	DNSName    string     `json:"name,omitempty" column:"name,width:30"`
+	// Key is the key used in Attach(). It's useful for users to
+	// understand where the event comes from.
+	Key string
```

Comment:
The JSON and Column tags are missing

Reply:
> It often contains the container id but it might not be the case when several containers share the same network namespace (e.g. sidecars in a pod, or containers with host netns).
>
@alban in which cases it might not?

IIUC, this approach differs from the ns/pod one because the packets will be displayed for every container (even if they share the same net ns). For instance, in the case of containers within the host netns, the same packet will be displayed for all of them. However, from my understanding, the same happens with the ns/pod approach (`kubeclt-gadget`) for two pods sharing the same netns, doesn't it?

If my understanding is correct, I was wondering if we could group by netns to make the output more clear, something like:

```bash
CONTAINER                  QR NAMESERVER         TYPE      QTYPE   NAME
cont1_with_netns_1         Q  10.96.0.10         OUTGOING  A       www.inspektor-gadget.io.
cont2_with_netns_1       
cont3_with_netns_1
cont1_with_netns_1         R  10.96.0.10         HOST      A       www.inspektor-gadget.io.
cont2_with_netns_1       
cont3_with_netns_1
cont4_with_netns_2         Q  10.96.0.10         OUTGOING  A       www.example.com.
cont4_with_netns_2         R  10.96.0.10         HOST  A           www.example.com.
```

What do you think? Same for `kubectl-gadget` but with ns and pod info. Maybe it would make the output clearer given that, with our implementation, we are not able to distinguish who is the "owner" of the packet among all the containers sharing the same netns.

Reply:
> It seems to me that Key is an internal implementation detail that should not be exposed.

My idea is to keep it internal. I thought that by missing the json tag it won't be marshaled but TIL that I have to use `json:"-"`. Also my comment `It's useful for users to understand where the event comes from.` is misleading, by users I meant developers. 

Ideally this field should be private, but types is a different package so we have to keep it public.